storage_controller/src/safekeeper.rs

```diff
@@ -57,9 +86,11 @@ impl Safekeeper {
         self.skp.scheduling_policy = scheduling_policy.into();
     }
     /// Perform an operation (which is given a [`SafekeeperClient`]) with retries
+    #[allow(clippy::too_many_arguments)]
     pub(crate) async fn with_client_retries<T, O, F>(
         &self,
         mut op: O,
+        http_client: reqwest::Client,
```

Comment:
nit: I think we can take a borrow here to avoid all of the cloning (this will require plumbing the borrow through to the callees too). It doesn't really matter for performance here, but I think it expresses the API contract more clearly, and is more convenient to use.